packages/shared/src/index.ts

```diff
@@ -124,7 +124,7 @@ export const capitalize = cacheStringFunction(
  * @private
  */
 export const toHandlerKey = cacheStringFunction((str: string) =>
-  str ? `on${capitalize(str)}` : ``
+  str ? `on${capitalize(camelize(str))}` : ``
```

Comment:
This is probably not the right place to fix it. It might be in the `emit` function: `toHandlerKey(camelize(event))` instead of `toHandlerKey(event)`

Reply:
well, looking at these test-cases:

- [trigger camelCase handler](https://github.com/vuejs/vue-next/blob/master/packages/runtime-core/__tests__/componentEmits.spec.ts#L37)
- [trigger kebab-case handler](https://github.com/vuejs/vue-next/blob/master/packages/runtime-core/__tests__/componentEmits.spec.ts#L55)
- [trigger mixed case handlers](https://github.com/vuejs/vue-next/blob/master/packages/runtime-core/__tests__/componentEmits.spec.ts#L74)

it seems that both, camel and kebab,  cases are supported - and if handlers are defined for both conventions, even both handlers should be invoked.
Might just be that the [warning](https://github.com/vuejs/vue-next/blob/master/packages/runtime-core/src/componentEmits.ts#L92) is wrong then? Should check weather at leas one of `['onFoo-bar', 'onFooBar']` is defined?

Reply:
Yeah, it could be the warning that is wrong. That's part of what I meant 😄